src/operations.js

```diff
@@ -111,25 +112,81 @@ export const optic = composeOptics
 export const path = composeOptics
 
 // reduce : Fold s a → (r -> a -> r) -> r -> s -> r
-export const reduce = curry((optic, f, i, obj) => optic.asFold.reduce(f, i, obj))
+export const reduce = curry((optic, f, i, obj) => {
+  if ('asFold' in optic) return optic.asFold.reduce(f, i, obj)
+  else
+    throw new UnavailableOpticOperationError(
```

Comment:
This looks interesting! Thanks! 🎉 
Can you please add a unit test per exception with the matcher `.shouldThrow(\Exception\)`?
This way our test will be a proof that badly constructed lenses should the appropiate error etc 😉

Reply:
You have them at the end of `operations.test.js`.